src/Features/Core/Portable/MakeFieldReadonly/AbstractMakeFieldReadonlyDiagnosticAnalyzer.cs

```diff
@@ -0,0 +1,114 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using Microsoft.CodeAnalysis.CodeStyle;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace Microsoft.CodeAnalysis.MakeFieldReadonly
+{
+    internal abstract class AbstractMakeFieldReadonlyDiagnosticAnalyzer<TConstructorDeclarationSyntax, TLambdaSyntax>
+        : AbstractCodeStyleDiagnosticAnalyzer
+        where TConstructorDeclarationSyntax : SyntaxNode
+        where TLambdaSyntax : SyntaxNode
+    {
+        private static readonly LocalizableString s_localizableTitle = new LocalizableResourceString(nameof(FeaturesResources.Add_readonly_modifier), FeaturesResources.ResourceManager, typeof(FeaturesResources));
+        private static readonly LocalizableString s_localizableMessage = new LocalizableResourceString(nameof(WorkspacesResources.Make_field_readonly), WorkspacesResources.ResourceManager, typeof(WorkspacesResources));
+
+        public AbstractMakeFieldReadonlyDiagnosticAnalyzer()
+            : base(IDEDiagnosticIds.MakeFieldReadonlyDiagnosticId,
+                   s_localizableTitle, s_localizableMessage)
+        {
+        }
+
+        public override bool OpenFileOnly(Workspace workspace) => false;
+
+        public override DiagnosticAnalyzerCategory GetAnalyzerCategory() => DiagnosticAnalyzerCategory.SemanticDocumentAnalysis;
+
+        internal void AnalyzeType(SyntaxNodeAnalysisContext context)
+        {
+            var typeSymbol = (ITypeSymbol)context.SemanticModel.GetDeclaredSymbol(context.Node);
+
+            var nonReadonlyFieldMembers = new HashSet<IFieldSymbol>();
```

Comment:
@CyrusNajmabadi Would using a `List<T>` fix this? It doesn't have to be a `HashSet<T>`.